driver/src/main/com/mongodb/client/MongoCollection.java

```diff
@@ -45,6 +45,12 @@
  *
  * <p>Note: Additions to this interface will not be considered to break binary compatibility.</p>
  *
+ * <p>MongoCollection is generic allowing for different types to represent {@code Documents}. Any custom classes must have a
+ * {@link org.bson.codecs.Codec} registered in the {@link CodecRegistry}. The default {@code CodecRegistry} includes built-in support for:
+ * {@link org.bson.BsonDocument}, {@link Document}, {@link com.mongodb.DBObject}. For more information see the Mongo Java Driver
+ * <a href="http://mongodb.github.io/mongo-java-driver/" target="_blank">documentation</a>.
```

Comment:
If we move the docs - we'd still have to redirect from the old domain - so that isnt really an issue.